packages/monaco/src/browser/monaco-frontend-module.ts

```diff
@@ -190,11 +178,34 @@ export function createMonacoConfigurationService(container: interfaces.Container
         }
         return sections;
     };
+
+    const toTarget = (scope: PreferenceScope): monaco.services.ConfigurationTarget => {
+        switch (scope) {
+            case PreferenceScope.Default: return monaco.services.ConfigurationTarget.DEFAULT;
+            case PreferenceScope.User: return monaco.services.ConfigurationTarget.USER;
+            case PreferenceScope.Workspace: return monaco.services.ConfigurationTarget.WORKSPACE;
+            case PreferenceScope.Folder: return monaco.services.ConfigurationTarget.WORKSPACE_FOLDER;
+        }
+    };
+
     preferences.onPreferencesChanged((changes?: PreferenceChanges) => {
         const affectedSections = parseSections(changes);
+
+        let target: monaco.services.ConfigurationTarget | undefined;
+        if (changes && affectedSections && affectedSections.length > 0) {
+            const change = changes[affectedSections[0]];
+            target = toTarget(change.scope);
```

Comment:
i am not sure that checking first is enough it would be safer to chunk changes by scope and fire for each scope with preserving changes order